src/cosmos/tx/actor.rs

```diff
@@ -0,0 +1,22 @@
+use crate::cosmos::tx::messages::trigger::TriggerTx;
+use crate::cosmos::tx::TxHandler;
+use actix::{Actor, AsyncContext, Context};
+use cosmrs::tx::Msg;
+use tracing::info;
+
+impl<T> Actor for TxHandler<T>
+where
+    T: Msg + Unpin + 'static,
+{
+    type Context = Context<Self>;
+
+    fn started(&mut self, ctx: &mut Self::Context) {
+        info!("📣 TxHandler started...");
+        ctx.run_interval(self.batch_window, |act, ctx| {
```

Comment:
If the transaction handler is responsible to handle the batching trigger, it can't be operated differently. By the way, as it handle the fees we can't adapt them according to the ongoing transactions.